src/canopy/llm/cohere.py

```diff
@@ -95,22 +105,33 @@ def chat_completion(self,
 
         connectors = model_params_dict.pop('connectors', None)
         messages: List[Dict[str, Any]] = self._map_messages(chat_history)
+        model_name = model_params_dict.pop('model', None) or self.model_name
 
         if not messages:
-            raise cohere.error.CohereAPIError("No message provided")
-
-        response = self._client.chat(
-            model=self.model_name,
-            message=messages.pop()['message'],
-            chat_history=messages,
-            documents=self.generate_documents_from_context(context),
-            preamble_override=system_prompt,
-            stream=stream,
-            connectors=[
-                {"id": connector} for connector in connectors
-            ] if connectors else None,
-            **model_params_dict
-        )
+            raise RuntimeError("No message provided")
+
+        if system_prompt:
+            messages = self._prepend_system_prompt_to_messages(system_prompt, messages)
```

Comment:
@scottmx81 @giannis2two I guess we can live with this solution for now.

I can think of some theoretical case when someone is using the `CohereLLM` class with a single instruction, without `context`. In this case sending his prompt as user message instead of `preamble_override` might  deteriorate the result, right?  
But that's totally an edge case for now, Canopy is mostly focused on RAG.

Reply:
Indeed this is valid. The delta of `preamble_override` vs "prompt as user message" should not have big impact for non-RAG behaviors. What we are really trying to optimize with this here is our RAG preamble that has noticeable impact when it is replaced.